src/lib/rmt/wizard_firewall_page.rb

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'y2firewall/firewalld'
+require 'ui/event_dispatcher'
+require 'rmt/utils'
+require "cwm/dialog"
+require "cwm/custom_widget"
+
+Yast.import 'CWMFirewallInterfaces'
+
+module RMT; end
+
+class RMT::WizardFirewallPage < CWM::Dialog
+  def initialize(config)
+    textdomain 'rmt'
+    @config = config
+  end
+
+  def title
+    'RMT configuration: Firewall'
+  end
+
+  def contents
+    HBox(
+      HStretch(),
+      VBox(
+        RemoteFirewall.new
+      ),
+      HStretch()
+    )
+  end
+
+  def run
+    if firewalld.installed?
+      firewalld.read
+
+      if Yast::Popup.AnyQuestion(
+        'Open firewall ports?',
+        "For RMT to work properly, the ports for HTTP (80) and HTTPS (443) need to be opened in the firewall.\nDo you want to open these ports now?",
+        'Yes', 'No', :yes
+      )
+        result = super
+        if result == :next
```

Comment:
you can write it as
`firewalld.write if result == :next`

Reply:
Yep, the whole if cascade is not needed in the end. Haven't yet cleaned up after myself. 
Probably I'll replace the whole `Popup.AnyQuestion` thing with simple text on the dialog itself, and just skip the dialog, when there is not firewall installed or enabled.

I just wanted to get the hard part done first :grimacing: